src/types/index.ts

```diff
@@ -368,16 +368,35 @@ export class StarknetContractFactory {
         this.metadataPath = config.metadataPath;
         this.casmPath = config.casmPath;
 
-        // find constructor
-        for (const abiEntryName in this.abi) {
-            const abiEntry = this.abi[abiEntryName];
-            if (isEntryAContructor(abiEntry, config.hre.config.paths, this.casmPath)) {
-                this.constructorAbi = <starknet.CairoFunction>abiEntry;
-                break;
-            }
-        }
+        const constructorPredicate = this.resolveContructorPredicate();
+        this.constructorAbi = findConstructor(this.abi, constructorPredicate);
     }
 
+    private resolveContructorPredicate(): (abiEntry: starknet.AbiEntry) => boolean {
+        if (!this.isCairo1()) {
+            return (abiEntry: starknet.AbiEntry): boolean => {
+                return abiEntry.type === "constructor";
+            };
+        }
+
+        const casmJson = JSON.parse(fs.readFileSync(this.casmPath, "utf-8"));
+        if (casmJson?.compiler_version.split(".")[0] !== "1") {
+            const msg = ".CASM json has to contain compiler_version '1.*.*'";
+            throw new StarknetPluginError(msg);
+        }
+
+        if (!casmJson?.entry_points_by_type?.CONSTRUCTOR) {
+            const msg = ".CASM json has to contain constructor entry point";
```

Comment:
change the message as it may be misleading

Reply:
Done